### PR TITLE
Handling file path on windows - 2

### PR DESCRIPTION
**Pull request by moritzraho**
continue on #44

### CODE REVIEW

**Comment by moritzraho** [comment]
@alexkli I've added a test

Regarding the other paths that are not joined, this is fine in windows, the nodejs runtime supports unix paths in every OS (there are even modules for converting all paths to unix e.g. https://www.npmjs.com/package/upath)

The problem is/was for windows paths in a linux env as it was the case when injecting the path source path into the linux container

File: src/kinds/nodejs/nodejs.js [hunk 1]
```diff
@@ -50,10 +51,15 @@ module.exports = {
         const bridgeSource = isCommonJS ? "mount-require.js" : "mount-plain.js";
 
         let code = fs.readFileSync(`${__dirname}/${bridgeSource}`, {encoding: 'utf8'});
+        let sourceFile = invoker.sourceFile.toString();
+
+        if(path.sep !== path.posix.sep) {
+          sourceFile = sourceFile.split(path.sep).join(path.posix.sep);
```
[the remark this answers is not in the document]
Comment (alexkli) [reply src/kinds/nodejs/nodejs.js]: Would be good to add a comment on what's going on here, such as
```
// On Windows, the path set on the cli would typically be in windows format,
// but the nodejs container is Unix and requires Unix paths
```

**Comment by alexkli** [comment]
Re-run worked. We should add a Windows circleci build if possible.

**Comment by alexkli** [comment]
This is the circleci windows documentation: https://circleci.com/docs/2.0/hello-world-windows/

We should try adding that, but one issue is probably that it does not support starting docker containers, which our tests need:

> Running Linux containers on Windows is not possible for now.

File: src/kinds/nodejs/nodejs.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (moritzraho) [reply src/kinds/nodejs/nodejs.js]: ok sure thanks for catching that (would be good to have eslint configured aswell at some point :) )

File: package.json [hunk 2]
```diff
@@ -18,8 +18,8 @@
         "wskdebug": "./cli.js"
     },
     "scripts": {
-        "pretest": "eslint .",
         "test": "nyc mocha test/**/*.test.js",
+        "posttest": "eslint .",
```
[the remark this answers is not in the document]
Comment (moritzraho) [reply package.json]: I like to use it.only to quickly iterate on one test + I have an auto test runner that runs npm test, however eslint complains if it.only is set and the test doesn't run

**Comment by moritzraho** [comment]
Thanks for the review Alex, if we cannot run linux containers it's a problem indeed, I guess in the future we could either wait for linux container support for windows or refactor the tests to use more mocks and not use docker

**Comment by moritzraho** [comment]
@alexkli I think we are good to go, wdyt? but some tests failed probably because of circle ci, I am not allowed to rerun those

**Comment by moritzraho** [comment]
Actually let's hold, I need to test this on a windows machine

**Comment by moritzraho** [comment]
Ok wasn't easy to get a windows environment with docker running, but all good, it works 👍

**Pull request merged**